README.md

```diff
@@ -55,29 +55,60 @@ The criteria for picking these event names included:
+
+For `testStart` and `testEnd`, the corresponding test object is passed to the reporter. The same applies to `suiteStart` and `suiteEnd` where the matching suite object is passed to the reporter. For `runStart` and `runEnd` a "global" suite object is passed to the reporter, which contains all top-level suites as child suites.
+
+The data structures are defined as follows:
+
+- **Suite**: `Object` - A suite is a collection of tests and potentially other suites.
+ - **name**: `String|undefined` - name of the suite, will be `undefined` only for the `globalSuite`
+ - **fullname**: `Array` - array of strings containing the name of the suite and the names of all its suites ancestors
+ - **childSuites**: `Array` - array with all direct subsuites
+ - **tests**: `Array` - array containing all tests that directly belong to the suite (but not to a child suite)
+ - **status**: `String|undefined` - summarized status of the suite, it will be `undefined` on the `suiteStart` event
+    - `failed`, if at least one test failed
+    - `skipped`, if all tests in the suite are skipped (and there is at least one skipped test)
+    - `passed`, if there is at least one passed test and all other tests are skipped or if there are no tests in the suite.
+ - **testCounts**: `Object` - contains how many tests have passed, failed etc. including the tests of child suites
+    - **passed**: `Number` - number of passed tests
+    - **failed**: `Number` - number of failed tests
+    - **skipped**: `Number` - number of skipped tests
+    - **total**: `Number` - total number of tests, the sum of the above 3 properties should equal this one
+ - **runtime**: `Number|undefined` - execution time of the whole suite in milliseconds (including child suites), it will be `undefined` on the `suiteStart` event
  
-For `testStart` and `testEnd`, the corresponding test object is passed to the reporter.  The same applies to `suiteStart` and `suiteEnd` where the matching suite object is passed to the reporter. For `runStart` and `runEnd` a "global" suite object is passed to the reporter, which contains all top-level suites as child suites.
-
-When `runStart`, `suiteStart` and `testStart` are emitted, the `status`, `runtime` and `errors` attributes are `undefined`.
+The above `suite properties` apply also for the `globalSuite`.
+
+- **Test**: `Object` - A test holds basic information on a single test/spec.
+ - **name**: `String` - name of the test
+ - **suiteName**: `String` - name of the suite the test belongs to
+ - **fullName**: `Array` - array of strings containing the name of the test and the names of all its suites ancestors
+ - **status**: `String|undefined` - result of the test, it will be `undefined` for the `testStart` event. Can be:
+    - `passed`, if all assertions have passed
+    - `failed`, if one assertion have failed
+    - `skipped`, a skipped test is disabled, i.e. it will not be executed
```

Comment:
To be consistent with the other two: "if the test is disabled and wasn't executed"?